src/DotNetWorker.Grpc/GrpcFunctionsHostLogger.cs

```diff
@@ -17,6 +19,8 @@ namespace Microsoft.Azure.Functions.Worker.Logging
     /// </summary>
     internal class GrpcFunctionsHostLogger : ILogger
     {
+        private static readonly ObjectSerializer _serializer = new JsonObjectSerializer();
```

Comment:
Instead of taking the serializer here -- have it injected via the constructor. That allows a customer to change it (see the Configuration sample in the repo for how they'd do that). They can change from Newtonsoft to System.Text.Json, for example.

Reply:
@brettsam Done! I followed your pattern of doing it through a set function instead of the constructor in `GrpcFunctionsHostLoggerProvider`, the same way `scopeProvider` is currently being set. Is that okay or should I change that?

Also just curious why `_scopeProvider` wasn't being set in the constructor the same way `_channelWriter` is when it is also a required parameter for `GrpcFunctionsHostLogger` and cannot be null.

Reply:
answered offline